File: VB/E232.Module.Win/MyDecimalCalcEditPropertyEditor.vb
Imports System
Imports DevExpress.Xpo
Imports DevExpress.XtraEditors
Imports DevExpress.Persistent.Base
Imports DevExpress.ExpressApp.Model
Imports DevExpress.ExpressApp.Editors
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.ExpressApp.Win.Editors
Imports DevExpress.XtraEditors.Repository

Namespace E232.Module.Win
    <PropertyEditor(GetType(Decimal), MyDecimalCalcEditPropertyEditor.CalcEditorAlias, False)> _
    Public Class MyDecimalCalcEditPropertyEditor
        Inherits DXPropertyEditor

        Public Const CalcEditorAlias As String = "CalcEditor"
        Public Sub New(ByVal objectType As Type, ByVal model As IModelMemberViewItem)
            MyBase.New(objectType, model)
            Me.ControlBindingProperty = "EditValue"
        End Sub
        Protected Overrides Function CreateControlCore() As Object
            Return New CalcEdit()
        End Function
        Protected Overrides Sub SetupRepositoryItem(ByVal item As RepositoryItem)
            MyBase.SetupRepositoryItem(item)
            CType(item, RepositoryItemCalcEdit).Mask.EditMask = "C"
            CType(item, RepositoryItemCalcEdit).Mask.UseMaskAsDisplayFormat = True
        End Sub
        Protected Overrides Function CreateRepositoryItem() As RepositoryItem
            Return New RepositoryItemCalcEdit()
        End Function
        ' Uncomment this code starting with version 13.1.
        'public override bool CanFormatPropertyValue {
        '    get { return true; }
        '}
    End Class
    <NavigationItem("E232")> _
    Public Class TestObject
        Inherits BaseObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        Private _Amount As Decimal
        <EditorAlias(MyDecimalCalcEditPropertyEditor.CalcEditorAlias)> _
        Public Property Amount() As Decimal
            Get
                Return _Amount
            End Get
            Set(ByVal value As Decimal)
                SetPropertyValue("Amount", _Amount, value)
            End Set
        End Property
    End Class
End Namespace

File: CS/E232.Module.Win/MyDecimalCalcEditPropertyEditor.cs
using System;
using DevExpress.Xpo;
using DevExpress.XtraEditors;
using DevExpress.Persistent.Base;
using DevExpress.ExpressApp.Model;
using DevExpress.ExpressApp.Editors;
using DevExpress.Persistent.BaseImpl;
using DevExpress.ExpressApp.Win.Editors;
using DevExpress.XtraEditors.Repository;

namespace E232.Module.Win {
    [PropertyEditor(typeof(decimal), MyDecimalCalcEditPropertyEditor.CalcEditorAlias, false)]
    public class MyDecimalCalcEditPropertyEditor : DXPropertyEditor {
        public const string CalcEditorAlias = "CalcEditor";
        public MyDecimalCalcEditPropertyEditor(Type objectType, IModelMemberViewItem model)
            : base(objectType, model) {
            this.ControlBindingProperty = "EditValue";
        }
        protected override object CreateControlCore() {
            return new CalcEdit();
        }
        protected override void SetupRepositoryItem(RepositoryItem item) {
            base.SetupRepositoryItem(item);
            ((RepositoryItemCalcEdit)item).Mask.EditMask = "C";
            ((RepositoryItemCalcEdit)item).Mask.UseMaskAsDisplayFormat = true;
        }
        protected override RepositoryItem CreateRepositoryItem() {
            return new RepositoryItemCalcEdit();
        }
        public override bool CanFormatPropertyValue {
            get { return true; }
        }
    }
    [NavigationItem("E232")]
    public class TestObject : BaseObject {
        public TestObject(Session session)
            : base(session) { }

        private decimal _Amount;
        [EditorAlias(MyDecimalCalcEditPropertyEditor.CalcEditorAlias)]
        public decimal Amount {
            get {
                return _Amount;
            }
            set {
                SetPropertyValue("Amount", ref _Amount, value);
            }
        }
    }
}